<launch>
    <node pkg="ros_viconstream" name="vicon" type="ros_viconstream_node" output="screen">
        <!-- Vicon url -->
        <param name="vicon_url" value="vicon-pc:801" />

        <!-- Vicon stream mode -->
        <!-- <param name="vicon_streammode" value="ClientPull" /> -->
        <param name="vicon_streammode" value="ServerPush" />

        <!-- Naming prefix of object -->
        <!-- <param name="object_prefix" value="myprefix" /> -->

        <!-- Name of the base "world" frame -->
        <param name="id_reference_frame" value="/world" />



        <!-- Object settings -->
        <!-- Example for an object named morph/morph -->

        <!-- Select frame for odometry -->
        <param name="morph/morph/global_frame_linear_velocity" value="false" />
        <param name="morph/morph/global_frame_angular_velocity" value="false" />

        <!-- Position calibration -->
        <param name="morph/morph/zero_pose/position/x" value="0" />
        <param name="morph/morph/zero_pose/position/y" value="0" />
        <param name="morph/morph/zero_pose/position/z" value="0" />

        <!-- Quaternion calibration -->
        <param name="morph/morph/zero_pose/rotation/w" value="1" />
        <param name="morph/morph/zero_pose/rotation/x" value="0" />
        <param name="morph/morph/zero_pose/rotation/y" value="0" />
        <param name="morph/morph/zero_pose/rotation/z" value="0" />

        <!-- RPY calibration, only quaternion OR RPY are allowed, not both. -->
        <!-- <param name="morph/morph/zero_pose/rotation/roll" value="0" /> -->
        <!-- <param name="morph/morph/zero_pose/rotation/pitch" value="0" /> -->
        <!-- <param name="morph/morph/zero_pose/rotation/yaw" value="0" /> -->

        <!-- Enable Occlusion errors -->
        <param name="morph/morph/errors/occluded" value="true" />

        <!-- Enable thresholds for frame to frame movement -->
        <param name="morph/morph/errors/max_delta_position" value="0.3" /> <!-- meters -->
        <param name="morph/morph/errors/max_delta_rotation" value="0.3" /> <!-- radians -->
    </node>
</launch>
